runtime/j9jitbuilder/JavaMethodCallbackGen.cpp

```diff
@@ -19,23 +19,303 @@
+		functionAddress = (void*) _vmThread->functions->CallShortMethodA;
+	} else if (_returnType == _vm->intReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallIntMethodA;
+	} else if (_returnType == _vm->longReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallLongMethodA;
+	} else if (_returnType == _objectClass) {
+		functionAddress = (void*) _vmThread->functions->CallObjectMethodA;
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+
+	return functionAddress;
+}
+
+char*
+JavaMethodCallbackGen::getCallInFunctionName()
+{
+	char* name = NULL;
+	if (_returnType == _vm->voidReflectClass) {
+		name = (char*) "CallVoidMethodA";
+	} else if (_returnType == _vm->booleanReflectClass) {
+		name = (char*) "CallBooleanMethodA";
+	} else if (_returnType == _vm->charReflectClass) {
+		name = (char*) "CallCharMethodA";
+	} else if (_returnType == _vm->floatReflectClass) {
+		name = (char*) "CallFloatMethodA";
+	} else if (_returnType == _vm->doubleReflectClass) {
+		name = (char*) "CallDoubleMethodA";
+	} else if (_returnType == _vm->byteReflectClass) {
+		name = (char*) "CallByteMethodA";
+	} else if (_returnType == _vm->shortReflectClass) {
+		name = (char*) "CallShortMethodA";
+	} else if (_returnType == _vm->intReflectClass) {
+		name = (char*) "CallIntMethodA";
+	} else if (_returnType == _vm->longReflectClass) {
+		name = (char*) "CallLongMethodA";
+	} else if (_returnType == _objectClass) {
+		name = (char*) "CallObjectMethodA";
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+	return name;
+}
+
+char*
+JavaMethodCallbackGen::getSignatureFromJ9Class(J9Class *type)
+{
+	char* name = NULL;
+	if (type == _vm->voidReflectClass) {
+		name = (char*) "V";
+	} else if (type == _vm->booleanReflectClass) {
+		name = (char*) "Z";
+	} else if (type == _vm->charReflectClass) {
+		name = (char*) "C";
+	} else if (type == _vm->floatReflectClass) {
+		name = (char*) "F";
+	} else if (type == _vm->doubleReflectClass) {
+		name = (char*) "D";
+	} else if (type == _vm->byteReflectClass) {
+		name = (char*) "B";
+	} else if (type == _vm->shortReflectClass) {
+		name = (char*) "S";
+	} else if (type == _vm->intReflectClass) {
+		name = (char*) "I";
+	} else if (type == _vm->longReflectClass) {
+		name = (char*) "J";
+	} else if (_returnType == _objectClass) {
+		name = (char*) "L";
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+	return name;
+}
+
+/**
+ * Line numbers of the JNI functions as defined in jni.h. These
+ * are used by DefineFunction(...)
+ */
+char*
+JavaMethodCallbackGen::getCallInFunctionLineNumber()
+{
+	char *line = NULL;
+	if (_returnType == _vm->voidReflectClass) {
+		line = (char*) "261";
+	} else if (_returnType == _vm->booleanReflectClass) {
+		line = (char*) "237";
+	} else if (_returnType == _vm->charReflectClass) {
+		line = (char*) "243";
+	} else if (_returnType == _vm->floatReflectClass) {
+		line = (char*) "255";
+	} else if (_returnType == _vm->doubleReflectClass) {
+		line = (char*) "258";
```

Comment:
Is there any way we can set up some sort of macro so we do not have to hardcode these values?  Can we modify jni.h to create these macros?  My suspicion is that we can / should not modify jni.h

Reply:
Yes, I don't think we can modify jni.h. @DanHeidinga would know more.

Is the line number essential? Can we ignore this?

Reply:
Line numbers are a debugging aid but not necessary to be accurate

Reply:
you can use __LINE__ if you like